datafusion-physical-expr/src/expressions/bitmap_distinct.rs

```diff
@@ -0,0 +1,229 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
[brainstorming] how about we combine this to `ApproxDistinct` and use `BitmapDistinctCountAccumulator` for int8, int16 and int32 if the feature is avilable?

And use `NumericHLLAccumulator` for int64 and other non-int types. This way, user just need declare `approx_distinct` and rely on Datafusion to auto select the best approximate algorithm

https://github.com/apache/arrow-datafusion/blob/81592947e8814327ebdbd1fbc3d4a090796e37a3/datafusion-physical-expr/src/expressions/approx_distinct.rs#L91-L98

---

unrelate notes: as a user, I do want to keep `count(distinct)` as exact count and `approx_distinct` as approximation